policy/manager.go

```diff
@@ -160,75 +161,87 @@ func (m *Manager) monitorPolicies(ctx context.Context, evalCh chan<- *sdk.Scalin
 
 			// Now send updates to the active policies or create new handlers
 			// for any new policies
-			for policyID, updated := range message.IDs {
+			err := m.processMessageAndUpdateHandlers(ctx, evalCh, message)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
 
-				updatedPolicy := &sdk.ScalingPolicy{}
-				var err error
+func (m *Manager) processMessageAndUpdateHandlers(ctx context.Context, evalCh chan<- *sdk.ScalingEvaluation, message IDMessage) error {
 
-				if !updated {
-					continue
-				}
+	for policyID, updated := range message.IDs {
 
-				s := m.policySources[message.Source]
-				updatedPolicy, err = s.GetLatestVersion(ctx, policyID)
-				if err != nil {
-					m.log.Error("encountered an error getting the latest version for policy", "policyID", policyID, "error", err)
-					if isUnrecoverableError(err) {
-						return err
-					}
-				}
+		updatedPolicy := &sdk.ScalingPolicy{}
+		var err error
+
+		if !updated {
+			continue
+		}
 
-				m.handlersLock.RLock()
-				pht := m.handlers[policyID]
-				m.handlersLock.RUnlock()
-				if pht != nil {
-					// If the handler already exists, send the updated policy to it.
-					m.log.Trace("sending updated policy to existing handler",
-						"policy_id", policyID, "policy_source", message.Source)
+		s := m.policySources[message.Source]
+		updatedPolicy, err = s.GetLatestVersion(ctx, policyID)
+		if err != nil {
+			m.log.Error("encountered an error getting the latest version for policy", "policyID", policyID, "error", err)
+			if isUnrecoverableError(err) {
+				return fmt.Errorf("failed to get latest version for policy %s: %w", policyID, err)
+			}
+			continue
```

Comment:
We're preventing the panic, but isn't this going to repeatedly log `encountered an error getting the latest version for policy` every time we have a policy ID for a policy without a [`policy`](https://developer.hashicorp.com/nomad/docs/job-specification/scaling#policy) map? Should we really be reporting that as an error at all?

Reply:
Or what am I not seeing?

Reply:
The issue described here https://github.com/hashicorp/nomad-autoscaler/issues/1124 was with a `scaling` block like the following:

```hcl
    scaling {
      enabled = true
      min     = 1
      max     = 10
    }
```

That's a scaling block without a `policy` map, which is valid for scaling without the Nomad Autoscaler and you'll get this error message (as the did the person who reported the issue). If there's no `policy` field should we always treat it as `enabled = false`?

Reply:
ok, so this is a different thing not directly related to the panic.

Reply:
> After running the exact example of the ticket, I still believe we need to show the error, because Nomad does not verify the job policy on submission, this is the only place were the policy is validated.

Well it's a valid policy, just not one the Autoscaler should be looking at. But I get your point here. Do we want to just tell users "hey if you _don't_ have a `policy` field then you should leave `enabled = false`" (the default)? That seems pretty reasonable to me, and we can then land this fix and a little docs adjustment to close this out.